app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperDownloader.kt

```diff
@@ -4,42 +4,42 @@
 
 package org.mozilla.fenix.wallpapers
 
-import android.content.Context
+import kotlinx.coroutines.CoroutineDispatcher
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.withContext
 import mozilla.components.concept.fetch.Client
 import mozilla.components.concept.fetch.Request
 import mozilla.components.concept.fetch.isSuccess
-import mozilla.components.support.base.log.logger.Logger
 import org.mozilla.fenix.BuildConfig
+import org.mozilla.fenix.wallpapers.Wallpaper.Companion.getLocalPath
 import java.io.File
 
 /**
  * Can download wallpapers from a remote host.
  *
- * @param context Required for writing files to local storage.
+ * @param filesDir The top level app-local storage directory.
  * @param client Required for fetching files from network.
+ * @param dispatcher Dispatcher used to execute suspending functions. Default parameter
+ * should be likely be used except for when under test.
  */
 class WallpaperDownloader(
-    private val context: Context,
+    private val filesDir: File,
```

Comment:
nit: thoughts?
```suggestion
    private val fileDirectory: File,
```

Reply:
`filesDir` matches the property name on `Context` such that `context.filesDir`. I'd still be okay changing it, but if we did I'd prefer to get even more descriptive, something like `storageRootDirectory` maybe?

Reply:
I totally forgot that was the name of the variable from `context`. your proposal, `storageRootDirectory`, looks good though!